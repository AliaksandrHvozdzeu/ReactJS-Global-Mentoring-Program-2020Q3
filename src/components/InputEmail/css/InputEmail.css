.input-email {
    line-height: 2rem;
    font-size: 1rem;
    background-color: #2E2E2E;
    color: var(--white);
    padding: 0.5rem 1rem;
    width: 100%;
    margin-top: 0.5rem;
}

.input-email-label {
    display: block;
    text-transform: uppercase;
    color: var(--red);
    font-size: 1.0rem;
    margin-bottom: 1.5rem;
}
