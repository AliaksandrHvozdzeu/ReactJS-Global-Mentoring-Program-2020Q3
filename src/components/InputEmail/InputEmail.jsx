import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const validate = () => {

};

const InputEmail = ({title }) => (
  <label htmlFor="login-email" className="input-email-label">
    {title}
    <input type="email"
           id="login-email"
           className="input-email"
           onChange={validate}/>
  </label>
);

InputEmail.propTypes = {
  title: PropTypes.string.isRequired
};

export default InputEmail;
