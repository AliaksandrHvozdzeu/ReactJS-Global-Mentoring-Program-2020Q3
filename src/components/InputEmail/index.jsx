import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const validate = () => {
  console.log("VALIDATE LOGIN");
};

const InputEmail = ({title }) => (
  <label htmlFor="login-email" className="input-email-label">
    {title}
    <input type="email"
           id="login-email"
           className="input-email"
           onChange={validate}/>
  </label>
);

InputEmail.propTypes = {
  title: PropTypes.string.isRequired
};

export default InputEmail;
