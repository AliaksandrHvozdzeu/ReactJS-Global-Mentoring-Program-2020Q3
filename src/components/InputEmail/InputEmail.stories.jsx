import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import InputEmail from './js/InputEmail';
import '../../App.css';

export default {
  title: 'InputEmail',
  decorators: [withKnobs],
};

export const Default = () => (
  <InputEmail />
);
