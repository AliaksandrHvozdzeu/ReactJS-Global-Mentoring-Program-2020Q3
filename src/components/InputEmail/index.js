import InputEmail from './js/InputEmail';

export default InputEmail;
