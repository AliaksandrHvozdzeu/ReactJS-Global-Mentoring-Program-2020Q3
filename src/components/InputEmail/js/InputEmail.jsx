import React from 'react';
import PropTypes from 'prop-types';
import '../css/InputEmail.css';

export default function InputEmail({ title }) {

  return (
    <>
      <label htmlFor="login-email" className="input-email-label">
        {title}
        <input type="email"
               id="login-email"
               className="input-email"/>
      </label>
    </>
  );

}

InputEmail.propTypes = {
  title: PropTypes.string.isRequired,
};
