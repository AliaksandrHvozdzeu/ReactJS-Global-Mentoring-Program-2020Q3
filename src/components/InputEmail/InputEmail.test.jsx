import React from 'react';
import rerender from 'react-test-renderer';
import InputEmail from './js/InputEmail';

describe('when Input', () => {
  test('then snapshot created', () => {
    const title = 'title';
    const component = rerender.create(<InputEmail title={title}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
