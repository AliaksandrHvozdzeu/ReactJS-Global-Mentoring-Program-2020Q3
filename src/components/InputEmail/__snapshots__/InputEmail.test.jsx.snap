// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Input then snapshot created 1`] = `
<label
  className="input-email-label"
  htmlFor="login-email"
>
  title
  <input
    className="input-email"
    id="login-email"
    type="email"
  />
</label>
`;
